snippets/c_api/iterate_events.cc

```diff
@@ -0,0 +1,181 @@
+/*******************************************************************************
+ The MIT License (MIT)
+
+ Copyright (c) 2015 Maps.Me
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+ *******************************************************************************/
+
+// This define is needed to preserve client's timestamps in events.
+#define ALOHALYTICS_SERVER
+#include "../../src/event_base.h"
+#include "../../../include/processor.h" // TODO: This is actually not from Alohalytics repo
+
+#include <algorithm>
+#include <iostream>
+#include <string>
+#include <set>
+
+#define DLLEXPORT extern "C"
+
+// Public struct used in Python
+struct UserInfo
+{
+  char os_t;
+  float lat;
+  float lon;
+  char raw_uid[32];
+};
+
+// Public struct used in Python
+struct EventTime
+{
+  uint64_t client_created;
+  uint64_t server_upload;
+};
+
+typedef void (*callbackFunction)(char const *, EventTime const &, UserInfo const &, char const **, int);
+
+// Helper function that convert KeyPairs event into list of (key, value) for use in Python
+void SetPairs(AlohalyticsKeyPairsEvent const * event, std::vector<char const *> & pairs)
```

Comment:
Generally speaking, it's *very* unsafe to store somewhere raw pointers to the content of C++ strings.